Create updated tavern yaml tests

Nevedrov opened the pull request:
-Created tavern tests for KUB-Super-Admin_API endpoints:
  -/admin
  -/admin/{id}
  -/admin/{id}/resend
-Updated requirements.txt to include tavern plugin
-Removed unnecessary python scripts
-Reworked folder structure of project

Nevedrov committed:
Update requirements.txt to include tavern 2.14.0 plugin

Nevedrov committed:
Create main tavern tests for KUB-Super-Admin_API endpoints

Nevedrov committed:
Create "bad request" response tests for KUB-Super-Admin_API endpoints

Nevedrov committed:
Create "unautorized" response tests for KUB-Super-Admin_API endpoints

Nevedrov committed:
Create "unprocessible entity" response tests for KUB-Super-Admin_API endpoints

Nevedrov committed:
Create "not found" response tests for KUB-Super-Admin_API endpoints

Nevedrov committed:
Create "user conflict" response test admins endpoint via POST method

Nevedrov committed:
Add finally blocks for clean up (delete created users) in tests

Nevedrov committed:
Merge test files for each endpoints into one

Nevedrov committed:
Remove unnecessary files

Nevedrov committed:
Update tests and stages descriptions

Nevedrov committed:
Move conftest.py from tests to API folder

Nevedrov committed:
-Add lacking admin/{id} bad request error get and put test
-Restore accidentally forgotten to merge admin/{id} unautorized error test
-Removed fixtures conflicting with tavern in conftest.py

d1mmm commented on KUB-Super-Admin_API/__init__.py:
KUB-Super-Admin_API should be a package not a folder

d1mmm commented on KUB-Super-Admin_API/tests/__init__.py:
tests/ should be a package

d1mmm commented on KUB-Super-Admin_API/tests/test_admins.tavern.yaml:
```diff
@@ -0,0 +1,271 @@
+    response:
+      status_code: 200
+      verify_response_with:
+        function: tavern.helpers:validate_pykwalify
```
write a comment about what this function does

Nevedrov committed:
chore:
-add KUB-Super-Admin_API and tests packages init files

Nevedrov committed:
chore:
-delete unnecessary fixtures in conftest.py

Nevedrov committed:
chore:
-add admins endpoint get method validate_pykwalify build in helper usage comment in test_admins.tavern.yaml

Nevedrov committed:
fix:
-removed wrong symbol in admins endpoint get method response validation schema

Nevedrov committed:
fix:
-test for unprocessable entity error via wrong id moved to not found error test

Nevedrov committed:
fix:
- test for unprocessable entity error via wrong id moved to not found error test
- added new test for unprocessable error via invalid request data

Nevedrov committed:
fix:
- delete test for unprocessable entity error for admin/{id}/resend post method

Nevedrov committed:
feat:
- changed request url to environment variable BASE_URL

Nevedrov committed:
chore:
- moved tavern tests from tests directory
- deleted tests folder

Nevedrov committed:
feat:
- add pytest hooks for logging request and response json input and output in tree-like format for each stage

Nevedrov committed:
refactor:
- update tests and stages names
- decouple test of POST and GET method for Unprocessible Entity error at /admins endpoint

Nevedrov committed:
refactor:
- remove repeating stages copy-paste using yaml anchors

Nevedrov committed:
feat: allure integration
- add makereport hook to conftest.py which replaces every tavern TestFailError with AssertionError for proper tests status in allure pages
- add supporting pytest fixtures for simplified access to repeated test stages names
- update requirements.txt to include allure-pytest
- update gitignore to exclude allure reports

Nevedrov committed:
refactor:
- decouple POST and PUT methods in positive cases test for admins/id endpoint into separate tests

Nevedrov committed:
fix:
- Remove TestFailError assignment

Nevedrov committed:
feat:
- add additional Bad Request and Not found error tests
- deleted unnecessary tests
- fixed minor mistakes in previous tests

Nevedrov committed:
refactor:
- update test names according to Behaviour-Driven-Design

Nevedrov committed:
refactor:
- remove minor mistakes in pytest hook

PavloKilko committed:
chore: rename dir, remove reduntand files

PavloKilko committed:
chore: rename dir, remove reduntand files
